snuba/web/query.py

```diff
@@ -166,6 +166,11 @@ def _format_storage_query_and_run(
     to collapse and disappear.
     """
 
+    # TODO: This function (well, it will be a wrapper of this function)
+    # where we will transform the result according to the SelectedExpression
+    # object in the query to ensure the fields in the QueryResult have
+    # the same name the user expects.
+
```

Comment:
I think this should be at this point of the query processing because:
1) After this function the query is formatted. No way to introduce an alias anymore in the clickhouse query. So here it is the last place where we have the original mapping to transform the result.
2) The query object itself should not exist beyond this stage (it is needed for historic reasons).
3) Doing this mapping before would be tricky because if we need to introduce aliases in the clickhouse query (to be able to do the mapping of the result reliably) we need to influence the formatting of the query when the ast will not include aliases anymore (that happens here)